File: backend/core-api/src/main/java/ai/govbiz/core/client/ai/AiSearchIntentPayload.java
package ai.govbiz.core.client.ai;

import java.util.List;

/**
 * AI Service의 내부 검색 의도 계약이다. Boolean wrapper는 누락된 필드와 false를 구분하기 위해
 * 의도적으로 사용한다.
 */
public record AiSearchIntentPayload(
        String originalQuery,
        List<String> keywords,
        List<String> regions,
        List<String> categories,
        List<String> targetTerms,
        Boolean acceptingOnly,
        Boolean clarificationNeeded,
        String clarificationQuestion,
        AiSearchIntentAnalysisMode analysisMode
) {
}

File: backend/core-api/src/main/kotlin/ai/govbiz/core/client/ai/AiSearchIntentPayload.kt
package ai.govbiz.core.client.ai

/**
 * AI Service의 내부 검색 의도 계약이다. Boolean nullable 타입은 누락된 필드와 false를
 * 구분하기 위해 의도적으로 사용한다.
 */
data class AiSearchIntentPayload(
    val originalQuery: String?,
    val keywords: List<String?>?,
    val regions: List<String?>?,
    val categories: List<String?>?,
    val targetTerms: List<String?>?,
    val acceptingOnly: Boolean?,
    val clarificationNeeded: Boolean?,
    val clarificationQuestion: String?,
)
